api/v1beta1/types_template.go

```diff
@@ -18,9 +18,286 @@ package v1beta1
 
 import (
 	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/utils/net"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
+// AzureManagedControlPlaneTemplateResourceSpec specifies an Azure managed control plane template resource.
+type AzureManagedControlPlaneTemplateResourceSpec struct {
+	// MachineTemplate contains information about how machines
+	// should be shaped when creating or updating a control plane.
+	// +optional
+	MachineTemplate *AzureManagedControlPlaneTemplateMachineTemplate `json:"machineTemplate,omitempty"`
+
+	// Version defines the desired Kubernetes version.
+	// +kubebuilder:validation:MinLength:=2
```

Comment:
```suggestion
	// +kubebuilder:validation:MinLength=2
```

Should there be a semicolon there? I searched through CAPZ and it seems like it's used both with and without the semicolon, so maybe try testing it out to see if the validation works.

Reply:
Since the non-template version of this field uses the colon, I thought to keep it consistent with that https://github.com/willie-yao/cluster-api-provider-azure/blob/clusterclass-aks-final/api/v1beta1/azuremanagedcontrolplane_types.go#L81